### PR TITLE
Check for successful autoloading after class is generated

### PR DESCRIPTION
**Pull request by ciaranmcnulty**
This is an FAQ when users have not correctly configured their autoloader and can get stuck in a 'Do you want to generate the class?' loop (see #585)

Now after a class is generated a quick `class_exists` checks whether the class has been written out to an autoloadable path and tells the user / cancels suite re-run if it is not the case.

autoloader

### CODE REVIEW

File: src/PhpSpec/Listener/ClassNotFoundListener.php [hunk 1]
```diff
@@ -75,7 +76,13 @@ public function afterSuite(SuiteEvent $event)
 
             if ($this->io->askConfirmation($message)) {
                 $this->generator->generate($resource, 'class');
-                $event->markAsWorthRerunning();
+                if (class_exists($classname)) {
+                    $event->markAsWorthRerunning();
+                }
+                else {
```
Comment (richardmiller) [inline src/PhpSpec/Listener/ClassNotFoundListener.php]: You could lose this else by throwing the exception if the class doesn't exist first.

File: src/PhpSpec/Listener/ClassNotFoundListener.php [hunk 2]
```diff
@@ -75,7 +76,13 @@ public function afterSuite(SuiteEvent $event)
...
+                    $message = 'File was written but the class %s was not autoloadable - do you have an autoloader configured?';
```
Comment (richardmiller) [inline src/PhpSpec/Listener/ClassNotFoundListener.php]: Is there more detail in the docs on this? If so, is it worth having the url for that in the message as well?

File: src/PhpSpec/Listener/ClassNotFoundListener.php (hunk 1 above)
Comment (ciaranmcnulty) [inline src/PhpSpec/Listener/ClassNotFoundListener.php]: nice, thanks

File: src/PhpSpec/Listener/ClassNotFoundListener.php (hunk 2 above)
Comment (ciaranmcnulty) [inline src/PhpSpec/Listener/ClassNotFoundListener.php]: There's not; I'll raise a documentation issue

**Pull request merged**